### ISSUE TITLE
add support for CancellationTokenSource and CancellationToken

### ISSUE THREAD

**Issue by prabirshrestha** [vim-jp/vital.vim#623]
CancellationTokenSource examples:
https://docs.microsoft.com/en-us/dotnet/standard/threading/cancellation-in-managed-threads
https://github.com/axios/axios

Abort Controllers: https://developer.mozilla.org/en-US/docs/Web/API/AbortController/AbortController

**Comment by lambdalisue** [comment]
Here you are. https://github.com/lambdalisue/vital-Whisky/tree/master/autoload/vital/__vital__/Async

**Comment by lambdalisue** [comment]
Yes but not yet because vital.vim does not run tests on Neovim yet.

**Comment by prabirshrestha** [comment]
I'm currently using [callbag.vim](https://github.com/prabirshrestha/callbag.vim) and cancellation is now very easy with `takeUntil`. The following will start logging when the user enters insert mode and automatically stops logging when the user leaves the insert mode.

```vim
     call callbag#pipe(
        \ callbag#fromEvent('TextChangedI'),
        \ callbag#takeUntil(
        \   callbag#fromEvent('InsertLeave'),
        \ ),
        \ callbag#debounceTime(250),
        \ callbag#subscribe({
        \   'next': {x->s:log('next')},
        \   'error': {x->s:log('error')},
        \   'complete': {->s:log('complete')},
        \ }),
        \ )
```

Feel free to close the issue.
